README.md

```diff
@@ -4,3 +4,207 @@
+  ~expected:"Robyn \"Rihanna\" Fenty"
+```
+
+By adding the `[@@deriving factory]` attribute to your type definitions, `ppx_factory` will generate
+a single factory function for record types or one per constructor for variant types.
+
+See the [Detailed usage](https://github.com/cryptosense/ppx_factory#detailed-usage) section for
+further specification.
+
+## Installation and usage
+
+You can install `ppx_factory` using [opam](https://opam.ocaml.org):
+```
+$ opam install ppx_factory
+```
+
+If you're building your library or app with dune, add the following field to your `library`,
+`executable` or `test` stanza:
+```
+(preprocess (pps ppx_factory))
+```
+or simply add `ppx_factory` to your `preprocess` field if it's already there.
+
+You can now add the `factory` and/or `default` plugins to `[@@deriving ...]` attributes on type
+definitions to derive the corresponding values.
+
+## Detailed usage
+
+### factory
+
+You can use `[@@deriving factory]` to derive factory functions from type definitions both in module
+and module signatures eg both in your `.ml` and `.mli` files given that it's an explicit record or
+variant type definition.
+
+If you try to derive factory functions from an abstract type, like in the following example:
+```ocaml
+type t1
+[@@deriving factory]
+
+type t2 = Some_external_module.t
+[@@deriving factory]
+```
+
+it will silently produce nothing. This is due to a current limitation in `ppxlib` combined with a
+necessity to be compatible with `ppx_import`. We're working on fixing that and hope to be able to
+provide users with an approriate error message in those cases instead.
+
+#### Record types
+
+You can derive factory functions from record type definitions. This will derive a single factory
+function that has an optional argument per field. The name of the factory function depends on the
+name of the type, `factory` will be derived from type `t` and `factory_<type_name>` for any other
+type.
+
+Each optional parameter will expect the same type as the one declared for the corresponding field,
+except for option types. A field which has type `a option` will be turned into a optional argument
+expecting an `a`.
+
+Because examples are worth a thousand words:
+```ocaml
+type t =
+  { a : int
+  ; b : string
+  }
+[@@deriving factory]
+
+type 'a u =
+  { c : 'a list
+  ; d : 'a option
+  }
+[@@deriving factory]
+```
+
+will derive the following factory functions:
+```ocaml
+val factory : ?a: int -> ?b: string -> unit -> t
+
+val factory_u : ?c: 'a list -> ?d: 'a -> unit -> 'a t
+```
+
+#### Variant types
+
+You can also derive factory functions from variant type definitions. This will derive one of them
+per constructor. Those functions will be named based on the type, just as for the record types but
+will have a `_<lowercased_constructor_name>` suffix.
+
+Constant constructors factories will have a single `unit` argument, while constructors with tuple
```

Comment:
`constructors` -> `constructor`?

Reply:
You're talking about the first one right? `Constant constructor factories` is indeed what I meant, I'll fix it!